packages/components/src/utils/getTransformedPageContent.ts

```diff
@@ -0,0 +1,41 @@
+import type { IsomerComponent } from "~/types"
+import { getDigestFromText } from "./getDigestFromText"
+import { getRandomNumberBetIntervals } from "./getRandomNumber"
+
+// if block.id is not present for heading level 2, we auto-generate one
+// for use in table of contents anchor links
+export const getTransformedPageContent = (content: IsomerComponent[]) => {
```

Comment:
food for thought: this seems to be catching hte case for prose blocks w/ content -> h2 that lack id. we can also add on studio side since tiptap can guarantee this property so that we can treat these similarly - wdyt?

Reply:
Hmm I was thinking of not letting studio handle this, because design is thinking of expanding this table of contents to also capture headings from other complex blocks. This guarantee is more important for components rather than on studio hence this check is here.

Reply:
ok, if this is for future proofing then sgtm!